cmd/admin-trace.go

```diff
@@ -378,11 +473,170 @@ func matchingOpts(ctx *cli.Context) (opts matchOpts) {
 		responseSize, e = humanize.ParseBytes(ctx.String("filter-size"))
 		fatalIf(probe.NewError(e).Trace(ctx.String("filter-size")), "Unable to parse input bytes.")
 	}
+
+	if ctx.String("rotate-trace-logs-size") != "" {
+		rotateTraceLogsSize, e = humanize.ParseBytes(ctx.String("rotate-trace-logs-size"))
+		fatalIf(probe.NewError(e).Trace(ctx.String("rotate-trace-logs-size")), "Unable to parse input trace logs bytes.")
+	}
+
+	if ctx.String("rotate-trace-logs-frequency") != "" {
+		rotateTraceLogsFrequency, e = ParseDuration(ctx.String("rotate-trace-logs-frequency"))
+		fatalIf(probe.NewError(e).Trace(ctx.String("rotate-trace-logs-frequency")), "Unable to parse input trace logs frequency.")
+	}
+
+	if ctx.String("rotate-trace-logs-pattern") != "" {
+		rotateTraceLogsPattern = ctx.String("rotate-trace-logs-pattern")
+		traceLogsPattern = rotateTraceLogsPattern
+	}
+
+	if ctx.String("rotate-trace-logs-compress") != "" {
+		rotateTraceLogsCompress = ctx.String("rotate-trace-logs-compress")
+		traceLogsCompress = rotateTraceLogsCompress
+	}
+
 	opts.requestSize = requestSize
 	opts.responseSize = responseSize
+	opts.rotateTraceLogsSize = rotateTraceLogsSize
+	opts.rotateTraceLogsFrequency = rotateTraceLogsFrequency
+	opts.rotateTraceLogsPattern = rotateTraceLogsPattern
+	opts.rotateTraceLogsCompress = rotateTraceLogsCompress
 	return
 }
 
+// Write trace log file
+func writeTracelogsFile(log string) {
+	traceLogsFileSize += len(log)
+	if _, err := traceLogsFile.WriteString(log + "\n"); err != nil {
+		traceLogsFile.Close()
+		fatalIf(probe.NewError(err), "Unable to write trace log file `"+defineTraceLogsFile()+"`")
+	}
+}
+
+// Compress trace log file
+func compressTraceLogsFile() {
+	// Create separate fd for asynch compress
+	switch traceLogsCompress {
+	case "gzip":
+		go compressGzip(traceLogsFile, traceLogsFilePath)
+	case "zstd":
+		go compressZstd(traceLogsFile, traceLogsFilePath)
+	default:
+		break
+	}
+}
+
+// gzip
+func compressGzip(asynchTraceLogsFile *os.File, asynchTraceLogsFilePath string) {
+	// Remove raw file
+	defer func() {
+		if err := os.Remove(asynchTraceLogsFilePath); err != nil {
+			// Ignore "no such file or directory". Another log archive already contains this data
+			if errors.Is(err, os.ErrNotExist) {
+				return
+			}
+			fatalIf(probe.NewError(err), "Unable to remove current trace log file `"+asynchTraceLogsFilePath+"`")
+		}
+	}()
+	compressFile, err := os.OpenFile(asynchTraceLogsFilePath+".gz", os.O_CREATE|os.O_RDWR, 0o644)
+	if err != nil {
+		compressFile.Close()
+		asynchTraceLogsFile.Close()
+		fatalIf(probe.NewError(err), "Unable to open trace log file archive for gzip `"+asynchTraceLogsFilePath+"`")
+	}
+	asynchTraceLogsFile.Seek(0, 0)
```

Comment:
```suggestion
	asynchTraceLogsFile.Seek(0, io.SeekStart)
```
Just to make it easier to read.